### PR TITLE
Add support for `categories` resource

### PR DESCRIPTION
**Pull request by dhartle4**
Support CRUD operations for BC `catalog/categories` resource. 

My Commands: 
``` python
bigc = BigCommerceCategoriesAPI('nice', 'try')
res_all = list(bigc.all())
res_get = bigc.get(24)
res_create = bigc.create(name='dillon')
res_update = bigc.update(category_id=29, data={'name':'dillon 10x', 'is_visible':False})
res_del = bigc.delete(29)
```
Responses:
image

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "bigc"
-version = "0.1.3"
+version = "0.1.4"
```
Comment (r-thomson) [inline pyproject.toml:3]: Let's leave this out of this PR so it can be in a separate commit

**Review by r-thomson — CHANGES_REQUESTED** [review]

**Comment by dhartle4** [comment]
classic FP

**Review by r-thomson — APPROVED** [review]

**Comment by maxheyn** [comment]
please take me off of this mailing list. i am sick of getting emails about this.

**Review by maxheyn — APPROVED** [review]
